tools/server/server.cpp

```diff
@@ -3691,6 +3698,77 @@ inline void signal_handler(int signal) {
     shutdown_handler(signal);
 }
 
+static bool check_pid_alive(const pid_t pid) {
+    if (pid <= 0) {
+        return false;
+    }
+
+    // Process is alive or exists but is inaccessible
+    if (kill(pid, 0) == 0 || errno == EPERM) {
+        return true;  // Process is alive
+    }
+
+    return false;  // Process does not exist or other error
+}
+
+class PidFile {
+  public:
+    FILE *      file = nullptr;
+    std::string fname;
+    bool        rm = false;
+
+    FILE * open(const std::string & filename, const char * mode, const bool r = false) {
+        file  = ggml_fopen(filename.c_str(), mode);
```

Comment:
we already had many functions to handle file i/o in `common.cpp`, should reuse it instead of inventing a new one